subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -699,6 +784,71 @@ static u16_t atou16(u8_t *buf, u16_t buflen, u16_t *len)
 	return val;
 }
 
+static int atos32(u8_t *buf, u16_t buflen, s32_t *val)
+{
+	bool negative = false;
+	u16_t pos = 0;
+
+	if (!buf || !buflen || !val) {
+		return -EINVAL;
+	}
+
+	*val = 0;
+
+	if (buf[pos] == '-') {
+		negative = true;
+		pos++;
+	} else if (buf[pos] == '+') {
+		pos++;
+	}
+
+	/* Compute val as a negative number to avoid overflow on INT_MIN */
+	while (pos < buflen && isdigit(buf[pos])) {
+		*val = *val * 10 - (buf[pos] - '0');
+		pos++;
+	}
+
+	if (!negative) {
+		*val = -*val;
+	}
+
+	return buflen == pos ? 0 : -EINVAL;
+}
+
+static int atof32(u8_t *buf, u16_t buflen, float32_value_t *val)
```

Comment:
Is this a simplified floating pointing notation without exponents, NaN, inf, et. al.?

Reply:
Yes, I believe so.
Detail is described at src/zephyr/include/net/lwm2m.h, line 137~152.

I think it should be sufficient for lwm2m usage.